[GTK][Flatpak] Allow filesystem access for a couple GStreamer debug variables

philn opened the pull request:
#### 952db1ae607e459d6044163d3fdadedc3a723cf4
<pre>
[GTK][Flatpak] Allow filesystem access for a couple GStreamer debug variables
<a href="https://bugs.webkit.org/show_bug.cgi?id=290544">https://bugs.webkit.org/show_bug.cgi?id=290544</a>

Reviewed by Michael Catanzaro.

If the user sets GST_DEBUG_FILE and/or GST_DEBUG_DUMP_DOT_DIR then the WebProcess needs filesystem
access to those locations. We use a similar approach in the bubblewrap launcher.

* Source/WebKit/UIProcess/Launcher/glib/FlatpakLauncher.cpp:
(WebKit::flatpakSpawn):

Canonical link: <a href="https://commits.webkit.org/292784@main">https://commits.webkit.org/292784@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7ccaa1927e6c325b02b93d981121d9f52459fc5e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/97070 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/16693 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/6901 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/102151 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/47595 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/16985 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/25143 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/73949 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31159 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/100073 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/12818 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/87795 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54286 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/12569 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/46925 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/82637 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/5730 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/104173 "Built successfully") | 
| | [❌ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/24144 "Hash 7ccaa192 for PR 43116 does not build (failure)") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/17614 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/82997 "Passed tests") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/24412 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/83903 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82393 "Passed tests") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/4631 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/17649 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15665 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/24109 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/29260 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/23932 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/27244 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/25505 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ntrrgc commented on Source/WebKit/UIProcess/Launcher/glib/FlatpakLauncher.cpp:
```diff
@@ -65,6 +66,21 @@ GRefPtr<GSubprocess> flatpakSpawn(GSubprocessLauncher* launcher, const WebKit::P
+            flatpakArgs.append(pathArg.get());
+        }
+
+        // GST_DEBUG_DUMP_DOT_DIR might not exist when the application starts, so we need write
```
What happens if the user running this doesn't have write permissions in the parent directory?

ntrrgc replied:
For instance, imagine you set up `GST_DEBUG_DUMP_DOT_DIR=/home/philn`. You have write permissions for `/home/philn` but not for `/home`.

philn replied:
Then nothing is written. These env vars are mostly for developers (and users wanting to report issues), so I would assume they have their permissions correctly set.

mcatanzaro reviewed (approved):
You can simplify the instructions on https://docs.webkit.org/Ports/WebKitGTK%20and%20WPE%20WebKit/Multimedia.html after this lands, right? The `--env="WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS=1"` is no longer needed?

philn committed:
[GTK][Flatpak] Allow filesystem access for a couple GStreamer debug variables
https://bugs.webkit.org/show_bug.cgi?id=290544

Reviewed by Michael Catanzaro.

If the user sets GST_DEBUG_FILE and/or GST_DEBUG_DUMP_DOT_DIR then the WebProcess needs filesystem
access to those locations. We use a similar approach in the bubblewrap launcher.

* Source/WebKit/UIProcess/Launcher/glib/FlatpakLauncher.cpp:
(WebKit::flatpakSpawn):

Canonical link: https://commits.webkit.org/292784@main
